packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -2503,6 +2530,8 @@ class LeaderLayer extends ContainerLayer {
         Matrix4.translationValues(offset.dx, offset.dy, 0.0).storage,
         oldLayer: _engineLayer as ui.TransformEngineLayer?,
       );
+    } else {
+      engineLayer = null;
```

Comment:
Nice catch. We should make this change even if we don't add `addToSceneWrapped`.